src/s3/explorer/s3BucketNode.ts

```diff
@@ -37,7 +37,7 @@ export class S3BucketNode extends AWSTreeNodeBase implements AWSResourceNode, Lo
     public constructor(
         public readonly bucket: Bucket,
         public readonly parent: S3Node,
-        private readonly s3: S3Client,
+        public readonly s3: S3Client,
```

Comment:
Do we need to make this public? Broadly speaking, you should be interacting with the public methods of the resource node to operate on them. If what you need doesn't exist, then add it as a new public method instead of bypassing the encapsulation.

If you are using the client but not operating on the node, then you should probably get the client from somewhere else.

Reply:
@daviddguez8 and I discussed this, it's so we can make all upload paths go through https://github.com/aws/aws-toolkit-vscode/pull/1799/files#diff-8e48e6040a121b730f20e563a29bd943e6fcbc935b4ab199343c6e760929d1eaR119 (the past iteration was tied directly to the node). Could consider pulling it from elsewhere, like making the client in this function and sending along the region for client creation (either from the node or from a setting like we have in the new workflow).

Reply:
Ah ok. I'm guessing the change to `uploadWithProgress` was done since the node (bucket) you're uploading to might not actually exist within the tree. 

This is OOS for this PR but I think this raises the bigger question of what should 'own' resources. Bypassing the explorer for CRUD(L) operations means the explorer is out of sync, and while yes you can just refresh it, it still seems rather odd (do we really need to list all buckets again after knowing what we just created?). Again, this is not something that this PR needs to implement or be concerned with, but something for us to think about.

Reply:
We don't have a way to access explorer nodes from outside the tree, which is definitely something we should explore (I was considering an overblown event subscription system at one point, emphasis on the overblown). That being said, there's never a guarantee that the explorer is up to date due to the fact that we will never know what is happening outside the scope of the toolkit...what happens if a user creates via the CLI, the console, etc.?

Reply:
Haha yeah I briefly though about using events as well but it's not something we really need. The issue of things happening outside the toolkit will always be relevant, and that's something we can't really solve on our own. However, for the things that we do have control over we should handle it nicely. Keeping the explorer up-to-date will always be best-effort, it's just that we can make a really good effort on our end versus basically nothing for things outside the toolkit.